portal/account/index_reset.php

```diff
@@ -0,0 +1,149 @@
+                        <td width="35%"><strong><?php echo xlt('Account Name'); ?><strong></td>
+                        <td><input class="form-control" name="uname" id="uname" type="text" readonly
+                                value="<?php echo attr($auth['portal_username']); ?>" /></td>
+                    </tr>
+                    <tr>
+                        <td><strong><?php echo xlt('New or Current Username'); ?><strong></td>
+                        <td><input class="form-control" name="login_uname" id="login_uname" type="text" required
+                                title="<?php echo xla('Change or keep current. Enter 8 to 20 characters'); ?>" pattern=".{8,20}"
+                                value="<?php echo attr($auth['portal_login_username']); ?>" />
+                        </td>
+                    </tr>
+                    <tr>
+                    <tr>
+                        <td><strong><?php echo xlt('Confirm Username'); ?><strong></td>
+                        <td><input class="form-control" name="confirm_uname" id="confirm_uname" type="text" required
+                                title="<?php echo xla('You must confirm this Username.'); ?>"
+                                autocomplete="none" pattern=".{8,20}" value="" />
+                        </td>
+                    </tr>
+                    </tr>
+                    <tr>
+                        <td><strong><?php echo xlt('New or Current Password'); ?><strong></td>
+                        <td>
+                            <input class="form-control" name="pass_new" id="pass_new" type="text" required
+                                placeholder="<?php echo xla('Min length is 8 with upper,lowercase,numbers mix'); ?>"
+                                title="<?php echo xla('You must enter a new or reenter current password to keep it.'); ?>"
+                                pattern="(?=.*\d)(?=.*[a-z])(?=.*[A-Z]).{8,}" />
+                        </td>
+                    </tr>
+                    <tr>
+                        <td><strong><?php echo xlt('Confirm Password'); ?><strong></td>
+                        <td>
+                            <input class="form-control" name="pass_new_confirm" id="pass_new_confirm" type="password"
+                                pattern="(?=.*\d)(?=.*[a-z])(?=.*[A-Z]).{8,}" autocomplete="none" />
+                        </td>
+                    </tr>
+                    <tr>
+                        <td colspan="2"><br><input class="btn btn-primary pull-right" type="submit" name="submit" value="<?php echo xla('Save'); ?>" /></td>
+                    </tr>
+                </table>
+                <div><?php echo '* ' . xlt("All credential fields are case sensitive!") ?></div>
+            </form>
+        <?php } elseif (isset($_POST['submit'])) {
+            if ($auth === false) {
+                unset($_POST['submit']);
+                header("Location: " . $_SERVER['PHP_SELF']);
+            }
```

Comment:
Shouldn't there be checks here that:
1. account name is correct
2. current password is correct

Also, something like this should probably do csrf prevention. I'll do some brainstorming on how to leverage the current OpenEMR\Common\Csrf\CsrfUtils to be used in the portal. I think it will work fine to simply call setupCsrfKey after successful login. I think :)

Reply:
Account name and login names are still being decided how to deal with but i'm looking to telemedicine. This is not finished yet but I needed to clear out my git projects as this commit has new files and I have not figured out a way to stash without removing the new files during stash.
If I don't manage the here and now well, frankly, i'll forget:):) This is not settled yet but i'll get there.
The questions also are being decided where to ask and whether I want to rotate or have canned 3. Maybe:
1. What is your SSN?
2. What is your CC#?
3. What is your CC pin? lol...
Really I need to decide if to force existing patients to reauth.
CSRF would be nice if you would help me out there. Meantime I need to take another pass at dialog to fix up cross origin and make the darn thing easier to maintain backwards compatibility.
The FHIR Questionnaire project is back on where openemr will send a Fhir ServiceRequest to another Fhir server to ask patient to complete the requested questionnaire(distant sends a push notification to patient) and return the response assessment where it may be eval'ed as measures etc. The project is related here because I have to write a SMART OpenID authentication/authorization server(Node or PHP!!!) so i'm hoping to get groundwork in portal to use the auth piece from Fhir...
awww, what was the question again!!! :)

Reply:
Sounds good. I'll attack the CSRF issue in the portal. Will be able to work on it in a couple days (I need to get out a patch first now that the more recent mariadb versions are getting use which do not work with original 5.0.2).